Content.Client/UserInterface/Systems/Alerts/AlertsUIController.cs

```diff
@@ -26,7 +26,17 @@ private void SystemOnClearAlerts(object? sender, EventArgs e)
     private void SystemOnSyncAlerts(object? sender, IReadOnlyDictionary<AlertKey, AlertState> e)
     {
         if (sender is ClientAlertsSystem system)
+        {
             UI?.SyncControls(system, system.AlertOrder, e);
+        }
+
+        // The UI can change underneath us if the user switches between HUD layouts
+        // So ensure we're subscribed to the AlertPressed callback.
+        if (UI != null)
+        {
+            UI.AlertPressed -= OnAlertPressed; // Ensure we don't hook into the callback twice
+            UI.AlertPressed += OnAlertPressed;
+        }
```

Comment:
Is there any reason this can't just be done on state exit? It looks hacky.

Reply:
Yeah, OnStateExited/OnStateEntered isn't called when switching the UI layout. The AlertsUIController is looking up a singleton whose instance can change - I don't see a suitable callback that can be hooked into when the value of UIManager.GetActiveUIWidgetOrNull<AlertsUI>() changes.